### PR TITLE
Fixed overflow function and added tests

### PR DESCRIPTION
**Pull request by bryanyli**

### CODE REVIEW

File: sunscreen_runtime/debugger-frontend/src/App.tsx [hunk 1]
```diff
@@ -248,96 +248,99 @@ const App = () => {
         setProblemNodes(problemNodes.concat([node.id]))
       }
     }
-    return newGraph
-  }, [session, problemNodes]
-  )
+    setGraph(newGraph)
```
Comment (samtay) [inline sunscreen_runtime/debugger-frontend/src/App.tsx:251]: I don't think you need to `setGraph` again here

File: sunscreen_runtime/src/debugger/data.rs [hunk 2]
```diff
@@ -313,3 +313,57 @@ pub struct ZkpNodeType {
...
 #[test]
 fn test_get_mul_depth() {}
+
+#[test]
+fn test_add_overflow() {
+    // Positive + Positive Overflow (6 + 4 > 7)
+    assert!(add_overflow_occurred(
+        [vec![vec![6, 5]], vec![vec![4, 6]]],
+        13
+    ));
```
Comment (samtay) [inline sunscreen_runtime/src/debugger/data.rs:323]: Can you put some comments on the `add_overflow_occurred` function? It's not clear to me how it works. For example here you have `(6, 5) + (4, 6) = (10, 11)` which should be fine with coefficient modulus `13`.

**Review by samtay — CHANGES_REQUESTED** [review]
Just a question on the add overflow logic before merging.

File: sunscreen_runtime/src/debugger/data.rs (hunk 2 above)
Comment (bryanyli) [reply sunscreen_runtime/src/debugger/data.rs:323]: The actual domain we're working with is [-p/2, p/2] where p is the coefficient modulus, so 6 + 4 = 10 which overflows the upper bound of 7.

Comment (samtay) [reply sunscreen_runtime/src/debugger/data.rs:323]: Gotcha. Yea you should add a doc comment to each of the overflow check functions explaining the arguments. And each of those functions should probably just be internal to this module, and not have the `pub` visibility. Push those changes and then merge it.

**Review by samtay — APPROVED** [review]
Doc tests would go above the actual functions, e.g. `add_overflow_occurred`, not the tests. But now that they're internal it's not quite as important.

**Pull request merged**